### DISCUSSION TITLE
How to pullup/pulldown signal in cocoTB

### DISCUSSION THREAD

**Discussion by suchi2108** [cocotb/cocotb#3187] [Q&A]
I am aware of 'cocotb.handle' library, and how force, deposit, freeze and release works.
I want to pull up the signal to '1'. I believe if it is pulled up, and I release it after some time, it will go to '1', instead of 'Z'.

We just write 'pullup(SDA)', where 'SDA' is wire type and it works fine in HDL. 

What could be substitute in CocoTB?

Any lead will be really helpful.

**Comment by ktbarrett** [comment]
There is no way to implement a pull up in cocotb, the procedural interfaces that cocotb uses do not support adding a new driver to a signal. You will have to do this in HDL. I suggest writing a simple pull up driver in HDL which cocotb controls with some external-facing signals.

**Reply by marlonjames** [reply]
There is a way to create drivers with VHPI and FLI, but it is not supported by cocotb, and I don't know the state of simulator support.

**Reply by ktbarrett** [reply]
Not supported by *all* PLIs that cocotb requires.
